internal/apis/configuration/v1/types.go

```diff
@@ -40,11 +40,12 @@ type Route struct {
 }
 
 type Upstream struct {
-	HashOn       string        `json:"hash_on"`
-	HashOnHeader string        `json:"hash_on_header"`
-	HashFallback string        `json:"hash_fallback"`
-	Healthchecks *Healthchecks `json:"healthchecks,omitempty"`
-	Slots        int           `json:"slots"`
+	HashOn          string        `json:"hash_on"`
+	HashOnHeader    string        `json:"hash_on_header"`
+	HashFallback    string        `json:"hash_fallback"`
+	Healthchecks    *Healthchecks `json:"healthchecks,omitempty"`
+	Slots           int           `json:"slots"`
+	ServiceUpstream bool          `json:"service_upstream"`
```

Comment:
Can we rename this to `UseClusterIP` to clearly convey the meaning of the property?